packages/airnode-node/src/coordinator/calls/chain-limits.ts

```diff
@@ -0,0 +1,61 @@
+import flatMap from 'lodash/flatMap';
+import groupBy from 'lodash/groupBy';
+import map from 'lodash/map';
+import zip from 'lodash/zip';
+import * as logger from '../../logger';
+import { Config, ProviderStates, PendingLog, GroupedRequests } from '../../types';
+import { flattenRequests, groupRequests } from '../../requests/grouping';
+import { sortRequests } from '../../requests/sorting';
+
+function flattenAndSortRequests(grouped: GroupedRequests) {
+  return sortRequests(flattenRequests(grouped));
+}
+
+export function applyChainLimits(config: Config, providerStates: ProviderStates) {
+  const logs: PendingLog[] = [];
+
+  const providerRequests = providerStates.evm
+    .map((evmState, index) => {
+      return { index, chainId: evmState.settings.chainId, requests: flattenAndSortRequests(evmState.requests) };
+    })
+    // Sort by number of request ascendingly
```

Comment:
"ascendingly" is a bit weird. Wasn't sure it was even a word (my browser's spell checker doesn't like it).

https://english.stackexchange.com/a/309294

```suggestion
    // Sort by number of request in an ascending order
```